src/libsyntax/ext/expand.rs

```diff
@@ -439,11 +440,13 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
             }
             ProcMacroDerive(..) | BuiltinDerive(..) => {
                 self.cx.span_err(attr.span, &format!("`{}` is a derive mode", attr.path));
+                self.cx.trace_macros_diag();
                 kind.dummy(attr.span)
             }
             _ => {
                 let msg = &format!("macro `{}` may not be used in attributes", attr.path);
                 self.cx.span_err(attr.span, msg);
+                self.cx.trace_macros_diag();
```

Comment:
Why do we need this here? This error shouldn't abort expansion.

Reply:
The order of the trace notes is unstable otherwise.